src/morphological_operations.jl

```diff
@@ -99,3 +99,85 @@ which is defined as the arithmetic difference between the internal and the exter
 `region` allows you to control the dimensions over which this operation is performed.
 """
 morpholaplace(img::AbstractArray, region=coords_spatial(img)) = dilate(img, region) + erode(img, region) - 2img
+
+abstract type ThinAlgo end
+
+doc"""
+    struct GuoAlgo <: ThinAlgo end
+
+A thinning operation algorithm.
+
+The procedure carries on until there are no changes between two consecutives iterations.
+
+The algorithm is described in the the page 361 (Algorithm A1) of:
+* Guo, Z., & Hall, R. W. (1989). Parallel thinning with two-subiteration algorithms. Communications of the ACM, 32(3), 359-373.
+"""
+struct GuoAlgo <: ThinAlgo end
+
+"""
+```
+function thinning_iteration(img::AbstractArray{Bool,2}; odd_iteration::Bool, algo::GuoAlgo)
+```
+The thining iteration evaluates three conditions in order to determine which pixels of the image should be removed from img.
+The three conditions are explained in the page 361 of:
+* Guo, Z., & Hall, R. W. (1989). Parallel thinning with two-subiteration algorithms. Communications of the ACM, 32(3), 359-373.
+"""
+function thinning_iteration!(img_ori::AbstractArray{Bool,2}, odd_iteration::Bool, algo::GuoAlgo)
```

Comment:
The `algo` option is not needed in `thinning_iteration`, you can get rid of it.

The documentation is much nicer now, thanks for the doc strings!